### PR TITLE
New Rule: Link: Multi-Factor Authentication phishing via QR code

### PR DESCRIPTION
**Pull request by morriscode**
@jkamdjou feel free to change the name! I wasn't sure about test rules, this uses message_screenshot and scan.qr

### CODE REVIEW

**Comment by jkamdjou** [comment]
> @jkamdjou feel free to change the name! I wasn't sure about test rules, this uses message_screenshot and scan.qr

i think we can test - @rw-access any objections?

**Comment by jkamdjou** [comment]
@morriscode there are some services that do send legit QR codes, eg im pretty sure Microsoft does, which would be unsolicited in most cases. do we need any additional signals like low rep sender or anything like that?

once we support proper URL extraction we can key off a sus link too, but sadly doesn't help us here yet

**Comment by morriscode** [comment]
@jkamdjou  totally agree I wanted to let this run in test and see if we could actually use signals to try to sus out what else to build on.  But maybe a cheaper rule would be better for that?

**Comment by jkamdjou** [comment]
> @jkamdjou  totally agree I wanted to let this run in test and see if we could actually use signals to try to sus out what else to build on.  But maybe a cheaper rule would be better for that?

awesome. i think unsolicited is probably enough to limit for testing, but will let @rw-access or @alexk307 confirm

File: detection-rules/link_multifactor_authentication_QR_code_phishing.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      (
+          regex.icontains(subject.subject, '(\b2fa\b|\bQ.?R\.?\s?\b|MFA|Muti[ -]?Factor Auth(entication)?)')
+      ),
+      (
+          regex.icontains(body.html.display_text, '(\b2fa\b|\bQ.?R\.?\s?\b|MFA|Muti[ -]?Factor Auth(entication)?)')
+      ),
+      (
+          regex.icontains(body.plain.raw, '(\b2fa\b|\bQ.?R\.?\s?\b|MFA|Muti[ -]?Factor Auth(entication)?)')
+      ),
+      (
```
Comment (rw-access) [inline detection-rules/link_multifactor_authentication_QR_code_phishing.yml:22]: idk if this `(` is standard but it seems pretty verbose to me. and `1 of` is technically `or` (gets turned into it anyway)

File: detection-rules/link_multifactor_authentication_QR_code_phishing.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  // QR code is present in attachment & recipient email address is present in the URL, a common tactic used in credential phishing attacks
+  and any(attachments,
+      .file_type in~ ('bmp', 'png', 'jpg', 'jpeg', 'gif')
+          and any(file.explode(.), .scan.qr.data is not null
+          and any(recipients.to, strings.icontains(..scan.qr.data, .email.email))
+      )
+  
+  // QR code is present in-line & recipient email address is present in the URL, a common tactic used in credential phishing attacks
+      or (
+          any(file.explode(beta.message_screenshot()), .scan.qr.data is not null
```
Comment (rw-access) [inline detection-rules/link_multifactor_authentication_QR_code_phishing.yml:42]: this should not be inside `any(attachments)`

File: detection-rules/link_multifactor_authentication_QR_code_phishing.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+              )
+          )
+      )
+  )
+  
+  // QR code is present in attachment & recipient email address is present in the URL, a common tactic used in credential phishing attacks
+  and any(attachments,
+      .file_type in~ ('bmp', 'png', 'jpg', 'jpeg', 'gif')
+          and any(file.explode(.), .scan.qr.data is not null
+          and any(recipients.to, strings.icontains(..scan.qr.data, .email.email))
```
Comment (rw-access) [inline detection-rules/link_multifactor_authentication_QR_code_phishing.yml:37]: indentation is off and I think it's messing up stuff below

**Review by rw-access — COMMENTED** [review]
I think it's filtered enough before the message screenshot, but there's one bug and some verbose formatting to tidy up

**Comment by morriscode** [comment]
Thanks @rw-access will revise in the morning!

File: detection-rules/link_multifactor_authentication_QR_code_phishing.yml (hunk 3 above)
Comment (morriscode) [reply detection-rules/link_multifactor_authentication_QR_code_phishing.yml:33]: I've indented and unintended it, it doesn't seem to make any difference in functionality. I'll update so the ands are inline.

**Comment by morriscode** [comment]
I lied tomorrow was so far away, I think it's how it should be now :)

File: detection-rules/link_multifactor_authentication_QR_code_phishing.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                    '(\b2fa\b|\bQ.?R\.?\s?\b|MFA|Muti[ -]?Factor Auth(entication)?)'))
+              )
+          )
+      )
+  )
+  
+  // QR code is present in attachment & recipient email address is present in the URL, a common tactic used in credential phishing attacks
+  and (
+      any(attachments,
+      .file_type in~ ('bmp', 'png', 'jpg', 'jpeg', 'gif')
```
Comment (jkamdjou) [inline detection-rules/link_multifactor_authentication_QR_code_phishing.yml:31]: this should be indented like you have on line 19

**Comment by morriscode** [comment]
Closing this going to resubmit a new variant

**Pull request closed without merging**